pkg/api/api.go

```diff
@@ -86,9 +87,10 @@ type API struct {
 	logger    log.Logger
 	sourceIPs *middleware.SourceIPExtractor
 	indexPage *IndexPageContent
+	limits    *validation.Overrides
 }
 
-func New(cfg Config, serverCfg server.Config, s *server.Server, logger log.Logger) (*API, error) {
+func New(cfg Config, serverCfg server.Config, s *server.Server, limits *validation.Overrides, logger log.Logger) (*API, error) {
```

Comment:
Note for reviewers: I've added `limits` as parameter here, because in the case of request rate limited error (new `RequestRateLimited`) the HTTP status code depends on the `-distributor.service-overload-status-code-on-rate-limit-enabled` CLI flag. The `limit` parameter is then passed to the constructor of the push handler.

Reply:
We don't need `limits` for API itself, only in `RegisterDistributor` method.

Reply:
And `RegisterIngester`.

Reply:
I have fixed this. Thank you.